Mathlib/NumberTheory/LSeries/Dirichlet.lean

```diff
@@ -0,0 +1,415 @@
+
+end zeta
+
+
+section vonMangoldt
+
+/-!
+### The L-series of the von Mangoldt function
+-/
+
+open LSeries Nat Complex ArithmeticFunction
+
+namespace ArithmeticFunction
+
+/-- A translation of the relation `Λ * ↑ζ = log` of (real-valued) arithmetic functions
+to an equality of complex sequences. -/
+lemma convolution_vonMangoldt_zeta : ↗Λ ⍟ ↗ζ = ↗Complex.log := by
+  ext n
+  simp only [zeta_apply, cast_ite, cast_zero, cast_one, LSeries.convolution_def, mul_ite,
+    mul_zero, mul_one]
+  simp_rw [← ofReal_zero, ← apply_ite, ← ofReal_sum]
+  norm_cast
+  simpa [-vonMangoldt_mul_zeta] using congr_arg (· n) vonMangoldt_mul_zeta
+
+lemma convolution_vonMangoldt_const_one : ↗Λ ⍟ 1 = ↗Complex.log :=
+  (convolution_one_eq_convolution_zeta _).trans convolution_vonMangoldt_zeta
+
+/-- The L-series of the von Mangoldt function `Λ` converges at `s` when `re s > 1`. -/
+lemma LSeriesSummable_vonMangoldt {s : ℂ} (hs : 1 < s.re) : LSeriesSummable ↗Λ s := by
+  have hs' : abscissaOfAbsConv 1 < s.re := by
+    rw [abscissaOfAbsConv_one]
+    exact_mod_cast hs
+  have hf := LSeriesSummable_logMul_of_lt_re hs'
+  rw [LSeriesSummable, ← summable_norm_iff] at hf ⊢
+  refine Summable.of_nonneg_of_le (fun _ ↦ norm_nonneg _) (fun n ↦ norm_term_le s ?_) hf
+  have hΛ : ‖↗Λ n‖ ≤ ‖Complex.log n‖ := by
+    simp only [norm_eq_abs, abs_ofReal, _root_.abs_of_nonneg vonMangoldt_nonneg,
+      ← Complex.natCast_log, _root_.abs_of_nonneg <| Real.log_nat_cast_nonneg n]
+    exact ArithmeticFunction.vonMangoldt_le_log
+  exact hΛ.trans <| by simp only [norm_eq_abs, norm_mul, Pi.one_apply, norm_one, mul_one, le_refl]
+
+end ArithmeticFunction
+
+namespace DirichletCharacter
+
+/-- A twisted version of the relation `Λ * ↑ζ = log` in terms of complex sequences. -/
+lemma convolution_twist_vonMangoldt {N : ℕ} (χ : DirichletCharacter ℂ N) :
+    (↗χ * ↗Λ) ⍟ ↗χ = ↗χ * ↗Complex.log := by
+  nth_rewrite 2 [← mul_one ↗χ]
+  rw [← DirichletCharacter.mul_convolution_distrib, convolution_vonMangoldt_const_one]
+
+/-- The L-series of the twist of the von Mangoldt function `Λ` by a Dirichlet character `χ`
+converges at `s` when `re s > 1`. -/
+lemma LSeriesSummable_twist_vonMangoldt {N : ℕ} (χ : DirichletCharacter ℂ N) {s : ℂ}
+    (hs : 1 < s.re) :
+    LSeriesSummable (↗χ * ↗Λ) s :=
+  LSeriesSummable_mul χ <| LSeriesSummable_vonMangoldt hs
+
+/-- The L-series of the twist of the von Mangoldt function `Λ` by a Dirichlet character `χ` at `s`
+equals the negative logarithmtic derivative of the L-series of `χ` when `re s > 1`. -/
+lemma LSeries_twist_vonMangoldt_eq {N : ℕ} (χ : DirichletCharacter ℂ N) {s : ℂ} (hs : 1 < s.re) :
+    L (↗χ * ↗Λ) s = - deriv (L ↗χ) s / L ↗χ s := by
+  rcases eq_or_ne N 0 with rfl | hN
+  · simp only [modZero_eq_delta, delta_mul_eq_smul_delta, vonMangoldt_apply_one, ofReal_zero,
+      zero_smul, LSeries_zero, Pi.zero_apply, LSeries_delta, Pi.one_apply, div_one, zero_eq_neg]
+    exact deriv_const s 1
```

Comment:
why not `simpa only ... using ...`?

Reply:
OK (20 heartbeats slower, though :smile:)